lib/wasi/src/lib.rs

```diff
@@ -133,138 +231,136 @@ impl WasiEnv {
+            "fd_pread" => Function::new_native_with_env(store, thread.clone(), fd_pread),
+            "fd_prestat_get" => Function::new_native_with_env(store, thread.clone(), fd_prestat_get),
+            "fd_prestat_dir_name" => Function::new_native_with_env(store, thread.clone(), fd_prestat_dir_name),
+            "fd_pwrite" => Function::new_native_with_env(store, thread.clone(), fd_pwrite),
+            "fd_read" => Function::new_native_with_env(store, thread.clone(), fd_read),
+            "fd_readdir" => Function::new_native_with_env(store, thread.clone(), fd_readdir),
+            "fd_renumber" => Function::new_native_with_env(store, thread.clone(), fd_renumber),
+            "fd_seek" => Function::new_native_with_env(store, thread.clone(), legacy::snapshot0::fd_seek),
+            "fd_sync" => Function::new_native_with_env(store, thread.clone(), fd_sync),
+            "fd_tell" => Function::new_native_with_env(store, thread.clone(), fd_tell),
+            "fd_write" => Function::new_native_with_env(store, thread.clone(), fd_write),
+            "path_create_directory" => Function::new_native_with_env(store, thread.clone(), path_create_directory),
+            "path_filestat_get" => Function::new_native_with_env(store, thread.clone(), legacy::snapshot0::path_filestat_get),
+            "path_filestat_set_times" => Function::new_native_with_env(store, thread.clone(), path_filestat_set_times),
+            "path_link" => Function::new_native_with_env(store, thread.clone(), path_link),
+            "path_open" => Function::new_native_with_env(store, thread.clone(), path_open),
+            "path_readlink" => Function::new_native_with_env(store, thread.clone(), path_readlink),
+            "path_remove_directory" => Function::new_native_with_env(store, thread.clone(), path_remove_directory),
+            "path_rename" => Function::new_native_with_env(store, thread.clone(), path_rename),
+            "path_symlink" => Function::new_native_with_env(store, thread.clone(), path_symlink),
+            "path_unlink_file" => Function::new_native_with_env(store, thread.clone(), path_unlink_file),
+            "poll_oneoff" => Function::new_native_with_env(store, thread.clone(), legacy::snapshot0::poll_oneoff),
+            "proc_exit" => Function::new_native_with_env(store, thread.clone(), proc_exit),
+            "proc_raise" => Function::new_native_with_env(store, thread.clone(), proc_raise),
+            "random_get" => Function::new_native_with_env(store, thread.clone(), random_get),
+            "sched_yield" => Function::new_native_with_env(store, thread.clone(), sched_yield),
+            "sock_recv" => Function::new_native_with_env(store, thread.clone(), sock_recv),
+            "sock_send" => Function::new_native_with_env(store, thread.clone(), sock_send),
+            "sock_shutdown" => Function::new_native_with_env(store, thread.clone(), sock_shutdown),
```

Comment:
`thread` is owned, so the last evaluated `Function` constructor could skip the `clone`. This depends on the order the macro evaluates its parameters, but the following suggestion might be the correct one:

```suggestion
            "sock_shutdown" => Function::new_native_with_env(store, thread, sock_shutdown),
```